ui/decredmaterial/editor.go

```diff
@@ -173,27 +149,56 @@ func (e Editor) Layout(gtx layout.Context) layout.Dimensions {
 	})
 }
 
-func (e Editor) editorLine(gtx C) D {
+func (e Editor) editorLayout(gtx C) D {
+	if e.Bordered {
+		return e.editorRectangle(gtx, func(gtx C) D {
+			return e.editorSection(gtx)
+		})
+	}
+
+	return e.editorSection(gtx)
+}
+
+func (e Editor) editorSection(gtx layout.Context) layout.Dimensions {
 	return layout.Flex{}.Layout(gtx,
+		layout.Flexed(1, func(gtx C) D {
+			return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					inset := layout.Inset{
+						Top:    e.m5,
+						Bottom: e.m5,
+					}
+					return inset.Layout(gtx, func(gtx C) D {
+						return e.EditorStyle.Layout(gtx)
+					})
+				}),
+			)
+		}),
 		layout.Rigid(func(gtx C) D {
-			dims := image.Point{
-				X: gtx.Constraints.Max.X,
-				Y: 2,
-			}
-			rect := f32.Rectangle{
-				Max: layout.FPt(dims),
+			if e.IsVisible {
+				inset := layout.Inset{
+					Top:  e.m2,
+					Left: e.m5,
+				}
+				return inset.Layout(gtx, func(gtx C) D {
+					if e.Editor.Text() == "" {
+						return e.pasteBtnMaterial.Layout(gtx)
+					}
+					return e.clearBtMaterial.Layout(gtx)
+				})
 			}
-			op.Offset(f32.Point{
-				X: 0,
-				Y: 0,
-			}).Add(gtx.Ops)
-			paint.ColorOp{Color: e.LineColor}.Add(gtx.Ops)
-			paint.PaintOp{Rect: rect}.Add(gtx.Ops)
-			return layout.Dimensions{Size: dims}
+			return layout.Dimensions{}
 		}),
 	)
 }
 
+func (e Editor) editorRectangle(gtx layout.Context, body layout.Widget) layout.Dimensions {
+	border := widget.Border{Color: e.LineColor, CornerRadius: e.m5, Width: unit.Dp(1)}
+	return border.Layout(gtx, func(gtx C) D {
+		return layout.Inset{Top: e.m2, Bottom: e.m2, Left: e.m5, Right: e.m5}.Layout(gtx, body)
+	})
+}
```

Comment:
is this method really needed? It seems to be used just once and the rest of the code is short enough to fit into the editorLayout method.